weaviate/collection/collection_model.py

```diff
@@ -416,36 +417,47 @@ def create(self, config: CollectionModelConfig[Model]) -> CollectionObjectModel[
         return CollectionObjectModel[config.model](self._connection, name, config.model)
 
     def get(self, model: Type[Model]) -> CollectionObjectModel[Model]:
-        name = _capitalize_first_letter(model.__name__)
-        path = f"/schema/{name}"
+        schema_props = self.__get_existing_properties(model)
 
-        try:
-            response = self._connection.get(path=path)
-        except RequestsConnectionError as conn_err:
-            raise RequestsConnectionError("Schema could not be retrieved.") from conn_err
-        if response.status_code != 200:
-            raise UnexpectedStatusCodeException("Get schema", response)
+        only_in_schema, only_in_model = self.__compare_properties_with_model(
+            schema_props, model.type_to_dict(model)
+        )
+        if len(only_in_schema) > 0 or len(only_in_model) > 0:
+            raise TypeError("Schemas not compatible")
+        return CollectionObjectModel[Model](
+            self._connection, _capitalize_first_letter(model.__name__), model
+        )
 
-        response_json = response.json()
-        model_props = model.type_to_dict(model)
-        schema_props = [
-            {"name": prop["name"], "dataType": prop["dataType"]}
-            for prop in response_json["properties"]
-        ]
+    def update_property(self, model: Type[Model]) -> CollectionObjectModel[Model]:
+        schema_props = self.__get_existing_properties(model)
+        name = _capitalize_first_letter(model.__name__)
+        only_in_schema, only_in_model = self.__compare_properties_with_model(
+            schema_props, model.type_to_dict(model)
+        )
+        if len(only_in_schema) > 0:
+            raise TypeError("Schema has extra properties")
 
-        def compare(s: List[Any], t: List[Any]) -> bool:
-            t = list(t)  # make a mutable copy
-            try:
-                for elem in s:
-                    t.remove(elem)
-            except ValueError:
-                return False
-            return not t
+        # we can only allow new optional types unless the default is None
+        for prop in only_in_model:
+            new_field = model.model_fields[prop["name"]]
+            non_optional_type = model.remove_optional_type(new_field.annotation)
+            if new_field.default is not None and non_optional_type == new_field.annotation:
+                raise WeaviateAddProperty(prop["name"])
+
+        for prop in only_in_model:
+            self.__add_property(name, prop)
 
-        if compare(model_props, schema_props):
-            raise TypeError("Schemas not compatible")
         return CollectionObjectModel[Model](self._connection, name, model)
 
+    def __add_property(self, name: str, additional_property: Dict[str, Any]):
+        path = f"/schema/{name}/properties"
```

Comment:
Should this have `__capitalize_first_letter`?

Reply:
I don't think it is neccesarry - I'd say capitalizing is the responsibility of the function that gets the user input, update_model in this case (just renamed this)